src/picknik_ur_site_config/objectives/open_cabinet_door.xml

```diff
@@ -0,0 +1,53 @@
+            <!-- Load the parameters needed for this behavior -->
+            <Action ID="LoadObjectiveParameters" config_file_name="open_door_fixed_handle_MTC_config.yaml" parameters="{parameters}"/>
+            <!-- Update the admittance controller and reset the FTS -->
+            <SubTree ID="Re-Zero Force-Torque Sensors" _collapsed="true"/>
+            <SubTree ID="Update Admittance Controller" _collapsed="true"/>
+
+            <!-- Get Poses from User -->
+            <Action ID="GetPoseFromUser" parameter_name="process_door_selection.hinge_axis_pose_end" parameter_value="{hinge_axis_pose_end}" />
+            <Action ID="GetPoseFromUser" parameter_name="process_door_selection.hinge_axis_pose_start" parameter_value="{hinge_axis_pose_start}" />
+            <Action ID="GetPoseFromUser" parameter_name="process_door_selection.grasp_pose" parameter_value="{grasp_pose}" />
+
+            <!-- Set up Subframes for MTC -->
+            <Action ID="GetMoveAlongArcSubframes" target_grasp_pose="{grasp_pose}" hinge_axis_pose_end="{hinge_axis_pose_end}" hinge_axis_pose_start="{hinge_axis_pose_start}" move_along_arc_subframes="{move_along_arc_subframes}" />
+            <Action ID="CreateGraspableObject" px="0.0" py="0.0" pz="0.0" rx="0.0" ry="0.0" rz="0.0" dx="0.1" dy="0.1" dz="0.1" frame_id="world" object_id="cabinet_door" generate_top_face="true" generate_front_face="true" generate_side_faces="true" cuboid_object="{door_object}"/>
+            <!-- Add the subframes to the Object-->
+            <Decorator ID="ForEachObjectSubframe" vector_in="{move_along_arc_subframes}" out="{subframe}">
+              <Control ID="Sequence">
+                <Action ID="AddSubframeToObject" subframe="{subframe}" graspable_object="{door_object}"/>
+              </Control>
+            </Decorator>
+            <!-- Open Gripper and begin MTC -->
+            <Action ID="MoveGripperAction" gripper_command_action_name="/robotiq_gripper_controller/gripper_cmd" position="0.0" />
+            <Control ID="Sequence" name="OpenDoorFixedHandleMTC">
+                <Action ID="InitializeMTCTask" task_id="open_door_fixed_handle_user" controller_names="/joint_trajectory_controller /robotiq_gripper_controller" task="{open_door_fixed_handle_task}" />
+                <Action ID="SetupMTCCurrentState" task="{open_door_fixed_handle_task}" />
+                <Action ID="SetupMTCGraspThenMoveAlongArcPull" graspable_object="{door_object}" parameters="{parameters}" task="{open_door_fixed_handle_task}" />
+                <Action ID="PlanMTCTask" solution="{open_door_fixed_handle_solution}" task="{open_door_fixed_handle_task}" />
+                <Fallback name="wait_for_approval_if_user_available">
+                    <Inverter>
+                        <Action ID="IsUserAvailable" />
+                    </Inverter>
+                    <Action ID="WaitForUserTrajectoryApproval" solution="{open_door_fixed_handle_solution}" />
+                </Fallback>
```

Comment:
Up to you, but for all these cases there is now a "Wait for Trajectory Approval If User Available" subtree that captures this entire logic. Might make the XMLs a little smaller.